More standard addVariable API pattern

dehann opened the pull request:
Why not do this, to make the API much closer to the existing standard?  

Here I just wrapped `addVariable` quickly.

dehann commented:
and it works:
```julia
julia> addVariable(nva, "x0", NVA.Pose2)
"01FZ4RBV58JYWR6R638W344F5W"

julia> 

```

jim-hill-r commented:
There is an entire plan for next gen API. We need to go through it in an engineering meeting. The goal is better alignment and solving for issues the Caesar API does not handle well. This is a can of worms targeted for after ICRA.

dehann commented:
Cool, we can keep this draft on ice till then

GearsAD commented:
Yes, the current API is clunky with the compound keys of client and navAbilityClient, but that's a 3-language refactoring. Much bigger story. 

Also I would like variable payloads to be trimmed down (e.g. not include `:default` solverDataDict), but that is part of the same discussion.

dehann commented:
okay, removed the review requests.  Added this to 'TBD API' project for future reference.  When we get around to standardization across SDKs, this will be a handy reference.  Going to close the PR in the mean time, but note that the suggested content has value for future discussions.

dehann commented:
> There is an entire plan for next gen API. We need to go through it in an engineering meeting. The goal is better alignment and solving for issues the Caesar API does not handle well. This is a can of worms targeted for after ICRA.

just noting current thoughts, still subject to larger design discussion:
- should we use uuids for factor names, which would allow cloud pipeline to remain stateless.  Jim opened a related issue to this somewhere on users choosing their own names.  Perhaps, but I'd default to uuid.
- Phantom variables with a "housekeeping" step before solve, which (based on customer config) automates phatom variable handling depending on use-case.  I think this will be critical to achieve good performance with async design.  It also means the `addFactor` call is going to get a few minor upgrades to simplify the entire experience.
